src/utils/CommonDefines.h

```diff
@@ -0,0 +1,30 @@
+/*
+ *      Copyright (C) 2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#if defined _WIN32 || defined __CYGWIN__
+  #define DLL_PRIVATE
+#else
+  #if __GNUC__ >= 4
+    #define DLL_PRIVATE  __attribute__ ((visibility ("hidden")))
+  #else
+    #define DLL_PRIVATE
+  #endif
+#endif
+
```

Comment:
For this is here https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h#L47 the ATTRIBUTE_HIDDEN

And under https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h#L284 set to CAddonBase